bspump/elasticsearch/data_feeder.py

```diff
@@ -0,0 +1,47 @@
+import orjson
+
+"""
+Data feeders to be used in ElasticSearchSink.
+"""
+
+
+async def data_feeder_create_or_index(items):
+	for _id, data in items:
+		yield b'{"create":{}}\n' if _id is None else orjson.dumps(
+			{"index": {"_id": _id}}, option=orjson.OPT_APPEND_NEWLINE
+		)
+		yield data
+
+
+async def data_feeder_create(items):
+	for _id, data in items:
+		yield b'{"create":{}}\n' if _id is None else orjson.dumps(
+			{"create": {"_id": _id}}, option=orjson.OPT_APPEND_NEWLINE
```

Comment:
I don't believe that the condition here makes sense ... it is condition before because it is create OR index => hence `if`.

In `create`, there is no condition - please consult the ElasticSearch manual - AFAIK you cannot specify `_id` for create.

Reply:
@ateska It is a condition, if ID is None or if it is specified, since it can be optional. You can specify ID in create:
https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html#docs-bulk-api-prereqs